### PR TITLE
[MU4] Add unit tests for change enharmonic spelling commands etc.

### PR DESCRIPTION
**Pull request by wizofaus**
Unit tests for previous PR

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [x] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/engraving/tests/tuplet_data/save-load.mscx [hunk 1]
```diff
@@ -214,7 +215,6 @@
               </Number>
             <numberType>1</numberType>
             <bracketType>1</bracketType>
-            <family>EB Garamond SC</family>
```
Comment (wizofaus) [inline src/engraving/tests/tuplet_data/save-load.mscx:217]: Not 100% sure why MU4 is removing this, but it looks ok...

**Comment by wizofaus** [comment]
Odd that those failing tests pass for me locally...they're not tests that are affected by this PR but I'd re-enabled them anyway as there didn't seem to be a reason not to.

File: src/engraving/tests/testbase.cpp [hunk 2]
```diff
@@ -245,5 +245,6 @@ void MTest::initMTest()
     root = rootPath();
     loadInstrumentTemplates(":/data/instruments.xml");
     score = readScore("test.mscx");
+    MScore::_error = Ms::MsError::MS_NO_ERROR;
```
Comment (wizofaus) [inline src/engraving/tests/testbase.cpp:248]: IMPORTANT! Without this many tests fail when run after tests such as timesig02 that set this.

File: src/framework/testing/qtestsuite.h [hunk 3]
```diff
@@ -91,7 +91,7 @@ class QTestSuite
         }
         int ret = 0;
         for (QObject* test : testList()) {
-            if (argc > 0 && test->objectName() == argv[1]) {
+            if (argc > 0 && std::find(argv + 1, argv + argc, test->objectName()) != argv + argc) {
```
Comment (wizofaus) [inline src/framework/testing/qtestsuite.h:94]: allows running engraving_tests with a list of class names (e.g. TestTimesig TestTools)

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**